pandas/tests/reshape/test_util.py

```diff
@@ -4,31 +4,60 @@
 import pandas.util.testing as tm
 from pandas.core.reshape.util import cartesian_product
 
+import string
+from datetime import date
+from dateutil import relativedelta
+
+from pandas.util._hypothesis import (st,
+                                     given,
+                                     settings,
+                                     get_seq,
+                                     assume)
+
+
+NO_OF_EXAMPLES_PER_TEST_CASE = 20
+
 
 class TestCartesianProduct(object):
 
-    def test_simple(self):
-        x, y = list('ABC'), [1, 22]
+    @settings(max_examples=NO_OF_EXAMPLES_PER_TEST_CASE)
+    @given(get_seq((str,), False, 1, 1),
```

Comment:
is there any way to name these things for ``given``, pretty non-intuitive here

Reply:
We are already giving them a name in some cases, as arguments in test case function. e.g
@given(st.lists(st.nothing()),
           get_seq((int,), False, min_size=1, max_size=10),
           get_seq((str,), False, min_size=1, max_size=10))
    def test_empty(self, **empty_list, list_of_int, list_of_str**):

local function can be added like below:
def get_str_list_with_single_element():
    return get_seq((str,), False, 1, 1)
but that would be too cumbersome. Instead,  for each argument in given decorator, having comments explaining each returned strategy can be more suitable.